lib/src/Timer/Timer.tsx

```diff
@@ -6,80 +6,67 @@ export type TimerProps = JSX.IntrinsicElements["div"] & {
     callback?: VoidFunction;
     /** Timer's duration in milliseconds */
     duration?: number;
+    /** Additional timer prefix */
+    timerPrefix?: React.ReactNode;
+    /** Additional timer suffix */
+    timerSuffix?: React.ReactNode;
 };
 
-interface TimerState {
-    timer: string;
-}
-/** A timer is a component for measuring time intervals */
-export class Timer extends React.Component<TimerProps, TimerState> {
-    private innerInterval: any;
-    constructor(props: TimerProps) {
-        super(props);
+export const Timer: React.FC<TimerProps> = ({ duration, callback, timerPrefix, timerSuffix, ...props }: TimerProps) => {
+    const [innerInterval, setInnerInterval] = React.useState<NodeJS.Timeout>();
+    const [timer, setTimer] = React.useState<string>("00:00");
 
-        this.state = {
-            timer: "00:00",
-        };
-    }
-
-    startInterval(timeout: number): void {
-        this.setState({ timer: this.convertMStoTime(timeout) }, () => {
-            this.clearInterval();
-            this.innerInterval = setInterval(() => {
+    const startInterval = (timeout: number): void => {
+        setTimer(formatMilitaryTime(timeout));
+        clearInterval();
+        setInnerInterval(
+            setInterval(() => {
                 if (timeout > 0) {
                     timeout = timeout - 1000;
-                    this.setState({ timer: this.convertMStoTime(timeout) }, () => {
-                        if (timeout === 0) {
-                            this.props.callback();
-                            this.clearInterval();
-                        }
-                    });
+                    setTimer(formatMilitaryTime(timeout));
+                    if (timeout === 0) {
+                        callback?.();
+                        clearInterval();
+                    }
                 }
-            }, 1000);
-        });
-    }
-
-    clearInterval(): void {
-        if (this.innerInterval) {
-            clearTimeout(this.innerInterval);
-            this.innerInterval = null;
-        }
-    }
-
-    convertMStoTime(value: number): string {
-        const date: Date = new Date(value);
-        return (
-            (date.getUTCHours() > 0 ? date.getUTCHours() + ":" : "") +
-            (date.getUTCMinutes() < 10 ? "0" + date.getUTCMinutes() : date.getUTCMinutes()) +
-            ":" +
-            (date.getUTCSeconds() < 10 ? "0" + date.getUTCSeconds() : date.getUTCSeconds())
+            }, 1000)
         );
-    }
+    };
 
-    componentDidMount() {
-        if (this.props.duration !== null && this.props.duration !== undefined) {
-            this.startInterval(this.props.duration);
+    const clearInnerInterval = (): void => {
+        if (innerInterval) {
+            clearInterval(innerInterval);
+            setInnerInterval(null);
         }
-    }
+    };
```

Comment:
tried to debug the timer and notice the interval is still running after the duration has ended, could not pin point the root cause of it; but was able to clear the intervals after merging all the helper functions into the side effects
```
    React.useEffect(() => {
        let intervalId;

        if (duration >= 1000) {
            // only start timer if there's at least 1 second set
            setTimer(convertMStoTime(duration));
            intervalId = setInterval(() => {
                if (duration > 0) {
                    duration -= 1000;
                    setTimer(convertMStoTime(duration));
                } else {
                    callback?.();
                    clearInterval(intervalId);
                }
            }, 1000);
            setInnerInterval(intervalId);
        }

        return () => clearInterval(intervalId);
    }, [duration]);
```

Reply:
yes :(  i cleaned up the code